test/unit/vagrant/util/directory_changed_test.rb

```diff
@@ -0,0 +1,40 @@
+require File.expand_path("../../../base", __FILE__)
+require "vagrant/util/directory_changed"
+require "time"
+
+describe Vagrant::Util::DirectoryChanged do
+  include_context "unit"
+
+  class FakePathname
+    attr_accessor :mtime
+    def initialize(mtime)
+      @mtime = mtime
+    end
+    def join(args)
+    end
+  end
+
+  let(:subject){ Vagrant::Util::DirectoryChanged }
+  
+  describe ".directory_changed?" do
+
+    let(:dir_paths) { ["/one/two.ext", "/one"] }
+    
+    before do
+      allow(Dir).to receive(:glob).and_return(dir_paths)
+      allow(Pathname).to receive(:new) do |arg|
+        FakePathname.new(Time.now)
```

Comment:
I suspect there is a better way to mock `Pathname` in order to set the mtime to something predictable to test against... 🤔